src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1892,7 +1893,8 @@ MulticopterPositionControl::task_main()
 									float k = thrust_xy_max / thrust_sp_xy_len;
 									thrust_sp(0) *= k;
 									thrust_sp(1) *= k;
-									saturation_xy = true;
+									/* Don't freeze x,y integrals if they both want to throttle down */
+									saturation_xy = ((vel_err(0) * _vel_sp(0) < 0.0f) && (vel_err(1) * _vel_sp(1) < 0.0f)) ? saturation_xy : true;
 								}
```

Comment:
We actually do have it. But you need to pull in the separate gains [PR](https://github.com/PX4/Firmware/pull/5555) for it.